We have a problem.
After the latest changes lkaino/Triflight:master still builds fine.
But with this PR it won't. It worked before the recent changes and for me its strange. The PR should make it smaller, not bigger.
Look here:

arm-none-eabi-gcc -o obj/main/triflight_NAZE.elf obj/main/NAZE/startup_stm32f10x_md_gcc.o obj/main/NAZE/drivers/accgyro_adxl345.o obj/main/NAZE/drivers/accgyro_bma280.o obj/main/NAZE/drivers/accgyro_l3g4200d.o obj/main/NAZE/drivers/accgyro_mma845x.o obj/main/NAZE/drivers/accgyro_mpu.o obj/main/NAZE/drivers/accgyro_mpu3050.o obj/main/NAZE/drivers/accgyro_mpu6050.o obj/main/NAZE/drivers/accgyro_mpu6500.o obj/main/NAZE/drivers/accgyro_spi_mpu6500.o obj/main/NAZE/drivers/adc.o obj/main/NAZE/drivers/adc_stm32f10x.o obj/main/NAZE/drivers/barometer_bmp085.o obj/main/NAZE/drivers/barometer_ms5611.o obj/main/NAZE/drivers/barometer_bmp280.o obj/main/NAZE/drivers/bus_spi.o obj/main/NAZE/drivers/bus_i2c_stm32f10x.o obj/main/NAZE/drivers/compass_hmc5883l.o obj/main/NAZE/drivers/display_ug2864hsweg01.o obj/main/NAZE/drivers/flash_m25p16.o obj/main/NAZE/drivers/gpio_stm32f10x.o obj/main/NAZE/drivers/inverter.o obj/main/NAZE/drivers/light_led_stm32f10x.o obj/main/NAZE/drivers/light_ws2811strip.o obj/main/NAZE/drivers/light_ws2811strip_stm32f10x.o obj/main/NAZE/drivers/sonar_hcsr04.o obj/main/NAZE/drivers/pwm_mapping.o obj/main/NAZE/drivers/pwm_output.o obj/main/NAZE/drivers/pwm_rx.o obj/main/NAZE/drivers/serial_softserial.o obj/main/NAZE/drivers/serial_uart.o obj/main/NAZE/drivers/serial_uart_stm32f10x.o obj/main/NAZE/drivers/sound_beeper_stm32f10x.o obj/main/NAZE/drivers/system_stm32f10x.o obj/main/NAZE/drivers/timer.o obj/main/NAZE/drivers/timer_stm32f10x.o obj/main/NAZE/io/flashfs.o obj/main/NAZE/hardware_revision.o obj/main/NAZE/flight/gtune.o obj/main/NAZE/flight/navigation.o obj/main/NAZE/flight/gps_conversion.o obj/main/NAZE/common/colorconversion.o obj/main/NAZE/io/gps.o obj/main/NAZE/io/ledstrip.o obj/main/NAZE/io/display.o obj/main/NAZE/telemetry/telemetry.o obj/main/NAZE/telemetry/frsky.o obj/main/NAZE/telemetry/hott.o obj/main/NAZE/telemetry/smartport.o obj/main/NAZE/telemetry/ltm.o obj/main/NAZE/sensors/sonar.o obj/main/NAZE/sensors/barometer.o obj/main/NAZE/blackbox/blackbox.o obj/main/NAZE/blackbox/blackbox_io.o obj/main/NAZE/build_config.o obj/main/NAZE/debug.o obj/main/NAZE/version.o obj/main/NAZE/config/config.o obj/main/NAZE/config/runtime_config.o obj/main/NAZE/common/maths.o obj/main/NAZE/common/printf.o obj/main/NAZE/common/typeconversion.o obj/main/NAZE/common/encoding.o obj/main/NAZE/common/filter.o obj/main/NAZE/scheduler.o obj/main/NAZE/scheduler_tasks.o obj/main/NAZE/main.o obj/main/NAZE/mw.o obj/main/NAZE/flight/altitudehold.o obj/main/NAZE/flight/failsafe.o obj/main/NAZE/flight/pid.o obj/main/NAZE/flight/imu.o obj/main/NAZE/flight/mixer.o obj/main/NAZE/flight/mixer_tricopter.o obj/main/NAZE/drivers/bus_i2c_soft.o obj/main/NAZE/drivers/serial.o obj/main/NAZE/drivers/sound_beeper.o obj/main/NAZE/drivers/system.o obj/main/NAZE/drivers/dma.o obj/main/NAZE/drivers/buf_writer.o obj/main/NAZE/drivers/gyro_sync.o obj/main/NAZE/io/beeper.o obj/main/NAZE/io/rc_controls.o obj/main/NAZE/io/rc_curves.o obj/main/NAZE/io/serial.o obj/main/NAZE/io/serial_1wire.o obj/main/NAZE/io/serial_cli.o obj/main/NAZE/io/serial_msp.o obj/main/NAZE/io/statusindicator.o obj/main/NAZE/rx/rx.o obj/main/NAZE/rx/pwm.o obj/main/NAZE/rx/msp.o obj/main/NAZE/rx/sbus.o obj/main/NAZE/rx/sumd.o obj/main/NAZE/rx/sumh.o obj/main/NAZE/rx/spektrum.o obj/main/NAZE/rx/xbus.o obj/main/NAZE/rx/ibus.o obj/main/NAZE/sensors/acceleration.o obj/main/NAZE/sensors/battery.o obj/main/NAZE/sensors/boardalignment.o obj/main/NAZE/sensors/compass.o obj/main/NAZE/sensors/gyro.o obj/main/NAZE/sensors/initialisation.o obj/main/NAZE/core_cm3.o obj/main/NAZE/system_stm32f10x.o obj/main/NAZE/stm32f10x_gpio.o obj/main/NAZE/stm32f10x_fsmc.o obj/main/NAZE/stm32f10x_exti.o obj/main/NAZE/stm32f10x_bkp.o obj/main/NAZE/stm32f10x_spi.o obj/main/NAZE/stm32f10x_adc.o obj/main/NAZE/stm32f10x_iwdg.o obj/main/NAZE/misc.o obj/main/NAZE/stm32f10x_wwdg.o obj/main/NAZE/stm32f10x_tim.o obj/main/NAZE/stm32f10x_usart.o obj/main/NAZE/stm32f10x_flash.o obj/main/NAZE/stm32f10x_rcc.o obj/main/NAZE/stm32f10x_sdio.o obj/main/NAZE/stm32f10x_i2c.o obj/main/NAZE/stm32f10x_pwr.o obj/main/NAZE/stm32f10x_rtc.o obj/main/NAZE/stm32f10x_dac.o obj/main/NAZE/stm32f10x_dbgmcu.o obj/main/NAZE/stm32f10x_dma.o -lm -nostartfiles --specs=nano.specs -lc -lnosys -mthumb -mcpu=cortex-m3 -flto -fuse-linker-plugin -Os -ggdb3 -DDEBUG -static -Wl,-gc-sections,-Map,./obj/main/triflight_NAZE.map -Wl,-L./src/main/target -Wl,--cref -T./src/main/target/stm32_flash_f103_128k.ld
c:/program files (x86)/gnu tools arm embedded/4.9 2015q3/bin/../lib/gcc/arm-none-eabi/4.9.3/../../../../arm-none-eabi/bin/ld.exe: obj/main/triflight_NAZE.elf section `.text' will not fit in region`FLASH'
c:/program files (x86)/gnu tools arm embedded/4.9 2015q3/bin/../lib/gcc/arm-none-eabi/4.9.3/../../../../arm-none-eabi/bin/ld.exe: region `FLASH' overflowed by 3620 bytes
collect2.exe: error: ld returned 1 exit status
Makefile:845: recipe for target 'obj/main/triflight_NAZE.elf' failed
make: **\* [obj/main/triflight_NAZE.elf] Error 1

--- a/src/main/flight/mixer_tricopter.c
+++ b/src/main/flight/mixer_tricopter.c
@@ -499,15 +499,17 @@ STATIC_UNIT_TESTED void tailTuneModeThrustTorque(thrustTorque_t *pTT, const bool
         if (!ARMING_FLAG(ARMED))
         {
             float averageServoAngle = pTT->servoAvgAngle.sum / 10.0f / pTT->servoAvgAngle.numOf;
-            if (averageServoAngle > 90.5f && averageServoAngle < 130.f) {
+            if (averageServoAngle > 90.5f && averageServoAngle < 120.f) {
                 averageServoAngle -= 90.0f;
                 averageServoAngle *= RAD;
-                gpMixerConfig->tri_tail_motor_thrustfactor = rint(10.0 / tan(averageServoAngle));
+                gpMixerConfig->tri_tail_motor_thrustfactor = 10.0f * cos_approx(averageServoAngle) / sin_approx(averageServoAngle);
 
                 saveConfigAndNotify();
 
                 pTT->state = TT_DONE;
-            } else {
+            }
+            else
+            {
                 pTT->state = TT_FAIL;
             }
             pTT->timestamp_ms = millis();

--- a/src/test/unit/mixer_tricopter_unittest.cc
+++ b/src/test/unit/mixer_tricopter_unittest.cc
@@ -73,6 +73,7 @@ protected:
             servo[i] = DEFAULT_SERVO_MIDDLE;
         }
 
+        mixerConfig.tri_tail_motor_thrustfactor = 123; // so we can check it's unchanged on TT_FAIL
         triInitMixer(&servoConf, &servo[5], &mixerConfig);
         tailTune.mode = TT_MODE_THRUST_TORQUE;
         tailTune.tt.state = TT_WAIT_FOR_DISARM;
@@ -118,6 +119,26 @@ TEST_F(ThrustFactorCalculationTest, 80) {
     // then
     EXPECT_NEAR(80, mixerConfig.tri_tail_motor_thrustfactor, 1);
     EXPECT_EQ(tailTune.tt.state, TT_DONE);
+}
+
+TEST_F(ThrustFactorCalculationTest, err90) {
+    // given
+    tailTune.tt.servoAvgAngle.sum = 270000;
+    // and
+    tailTuneModeThrustTorque(&tailTune.tt, true);
+    // then
+    EXPECT_EQ(123, mixerConfig.tri_tail_motor_thrustfactor);
+    EXPECT_EQ(tailTune.tt.state, TT_FAIL);
+}
+
+TEST_F(ThrustFactorCalculationTest, err130) {
+    // given
+    tailTune.tt.servoAvgAngle.sum = 390000;
+    // and
+    tailTuneModeThrustTorque(&tailTune.tt, true);
+    // then
+    EXPECT_EQ(123, mixerConfig.tri_tail_motor_thrustfactor);
+    EXPECT_EQ(tailTune.tt.state, TT_FAIL);
 }
 
 //STUBS

